### ISSUE TITLE
Issue in Get JSON with Jquery getJSON method challenge

### ISSUE THREAD

**Issue by arun1595** [FreeCodeCamp/FreeCodeCamp#8543]
#### Challenge Name
Get JSON with the jQuery getJSON Method
[Challenge Link](https://www.freecodecamp.com/challenges/get-json-with-the-jquery-getjson-method)

#### Issue Description
Eventhough my solution is right, the tests don't pass.

#### Browser Information

* Browser Name, Version: Google Chrome 50.0.2661.94 (Official Build) (64-bit) 
* Operating System: Linux
* Mobile, Desktop, or Tablet: Laptop

#### My Code

```
$("#getMessage").on("click", function(){
      
      $.getJSON("/json/cats.json", function(data){
        $(".message").html(JSON.stringify(data));
      });
      
});
```

#### Screenshot
![solution_not_working](https://cloud.githubusercontent.com/assets/7039523/15132356/693cea32-1676-11e6-95ac-3fd47bba53e1.png)

**Comment by raisedadead** [comment]
@arun1595 Thanks for reporting this.

Looks like it is checking for the var named `json` instead of being generic like you have used `data`.
I think the tests should allow flexibility.

**Comment by ttymed** [comment]
Hihi,
I'd like to work on this.

**Comment by raisedadead** [comment]
@ttymed yup sure!

**Comment by Bouncey** [comment]
If I'm honest, the `.html(JSON.stringify(JSON)` had me confused as it was my first experience working with them. I had to refer back a number of times whilst building my weather app to see just what was going on. 

Would a possible solution be to regEx `$('.message')` for either `!== "the message will go here` or `=== (the data from)JSON`?

**Comment by ttymed** [comment]
I don't quite understand your meaning.
The problem with the test right now is that you have to use the word json to define the parameter in the callback function and the .htmlJSON.stringify(json) excerpt.
My understanding was that the test should be more flexible so that the parameter and input in the .html function can use any word, just like in regular Javascript code.

**Comment by raisedadead** [comment]
@ttymed You have the correct understanding.
@Bouncey What you are saying is not in the scope for this challenge. To further clarify, this challenge aims to teach campers use of getting information with the `getJSON` API.

**Comment by Bouncey** [comment]
@raisedadead sorry, I meant in the testing of the campers script, not as part of the challenge.

**Comment by Bouncey** [comment]
@raisedadead 
I have made this line more flexible
```
$.getJSON("/json/cats.json", function(json) {
```
`function(json)` can now be `function(data)` or `function(whateverYouLike)`. I accomplished this by changing the testing regex slightly.

I am having problems with this line though
```
$(".message").html(JSON.stringify(data));
```
If I try any regex changes, it gets fouled up with false positives or false negatives depending on what I have done.

would there be a way of querying if $('.message') matches the first object of the JSON? Are there any other attributes to the `assert` method used for testing other than `code.match`?
